VEDA_OS01/transcripts.py

```diff
@@ -850,109 +942,112 @@ def retrieve_three_play_translations():
     finally, update it in edx-val.
     """
     log_prefix = u'3PlayMedia Task'
-    translation_processes = TranscriptProcessMetadata.objects.filter(
-        provider=TranscriptProvider.THREE_PLAY,
-        status=TranscriptStatus.IN_PROGRESS,
-    ).exclude(Q(translation_id__isnull=True) | Q(translation_id__exact=''))
-
-    for translation_process in translation_processes:
-
-        log_args = (
-            translation_process.video.studio_id,
-            translation_process.lang_code,
-            translation_process.process_id,
-        )
+    candidate_videos = Video.objects.filter(
+        provider=TranscriptProvider.THREE_PLAY, transcript_status=TranscriptStatus.IN_PROGRESS,
+    )
+    for video in candidate_videos:
+        # For a video, fetch its in progress translation processes.
+        in_progress_translation_processes = get_in_progress_translation_processes(video)
+        if not in_progress_translation_processes.exists():
+            LOGGER.info(
+                '[3PlayMedia Task] video=%s does not have any translation process who is in progress.',
+                video.studio_id,
+            )
+            continue
 
-        course_id = translation_process.video.inst_class.local_storedir.split(',')[0]
+        # Process id remains same across all the processes of a video and its also referred as `file_id`.
+        file_id = in_progress_translation_processes.first().process_id
+        course_id = video.inst_class.local_storedir.split(',')[0]
         org = utils.extract_course_org(course_id=course_id)
 
         # Retrieve transcript credentials
         three_play_secrets = get_transcript_credentials(
             provider=TranscriptProvider.THREE_PLAY,
             org=org,
-            edx_video_id=translation_process.video.studio_id,
-            file_id=translation_process.process_id,
+            edx_video_id=video.studio_id,
+            file_id=file_id,
             log_prefix=log_prefix
         )
         if not three_play_secrets:
-            # Fail the process
-            translation_process.update(status=TranscriptStatus.FAILED)
+            in_progress_translation_processes.update(status=TranscriptStatus.FAILED)
             continue
 
-        # Check transcript status
-        translation_status = get_translation_status(
-            three_play_secrets.api_key,
-            translation_process.process_id,
-            translation_process.translation_id,
-            translation_process.video.studio_id,
-            translation_process.lang_code,
+        # Retrieve Translations metadata to check the status for each translation.
+        translations = get_translations_metadata(
+            api_key=three_play_secrets.api_key,
+            file_id=file_id,
+            edx_video_id=video.studio_id,
```

Comment:
I would extract translation logic into new method. Method with 125 lines won't be acceptable.